src/main/java/com/bitvavo/api/Bitvavo.java

```diff
@@ -189,21 +137,21 @@ public void errorRateLimit(JSONObject response) {
       String placeHolder = message.split(" at ")[1].replace(".", "");
       rateLimitReset = Long.parseLong(placeHolder);
       if (!rateLimitThreadStarted) {
-        new Thread(new Runnable() {
-          public void run() {
-            try {
-              long timeToWait = rateLimitReset - System.currentTimeMillis();
-              rateLimitThreadStarted = true;
-              debugToConsole("We are waiting for " + ((int) timeToWait / (int) 1000) + " seconds, untill the rate limit ban will be lifted.");
+        new Thread(() -> {
+          try {
+            long timeToWait = rateLimitReset - System.currentTimeMillis();
+            rateLimitThreadStarted = true;
+            debugToConsole("We are waiting for " + ((int) timeToWait / 1000) + " seconds, until the rate limit ban will be lifted.");
+            if (timeToWait > 0) {
```

Comment:
I would appreciate is this is merged ASAP

Reply:
@jctim I'd recommend reaching out to Bitvavo customer service. I've tried to let them now that this merge request has been open for quite some time, but had no success. Perhaps someone else contacting them might help. For a company the size of Bitvavo having code this shitty on their GitHub is an embarrassment.